lsproxy/src/ast_grep/client.rs

```diff
@@ -1,50 +1,100 @@
 use tokio::process::Command;
 
 use super::types::AstGrepMatch;
+use crate::api_types::{get_mount_dir, FilePosition, Position, Symbol};
 
 pub struct AstGrepClient {
-    pub config_path: String,
+    pub symbol_config_path: String,
+    pub identifier_config_path: String,
+    pub reference_config_path: String,
 }
 
 impl AstGrepClient {
-    pub async fn get_file_symbols(
+    pub async fn get_symbol_from_position(
         &self,
         file_name: &str,
-    ) -> Result<Vec<AstGrepMatch>, Box<dyn std::error::Error>> {
-        let command_result = Command::new("ast-grep")
-            .arg("scan")
-            .arg("--config")
-            .arg(&self.config_path)
-            .arg("--json")
-            .arg(file_name)
-            .output()
+        identifier_position: &lsp_types::Position,
+    ) -> Result<Symbol, Box<dyn std::error::Error>> {
+        // Get all symbols in the file
+        let full_path = get_mount_dir().join(&file_name);
+        let full_path_str = full_path.to_str().unwrap_or_default();
+        let file_symbols = self
+            .scan_file(&self.symbol_config_path, full_path_str)
             .await?;
 
-        if !command_result.status.success() {
-            let error = String::from_utf8_lossy(&command_result.stderr);
-            return Err(format!("sg command failed: {}", error).into());
+        // Find the symbol that matches our identifier position
+        let symbol_result = file_symbols.into_iter().find(|symbol| {
+            symbol.range.start.line == identifier_position.line
+                && symbol.range.start.column == identifier_position.character
+        });
+        match symbol_result {
+            Some(matched_symbol) => Ok(Symbol::from(matched_symbol)),
+            None => Err(Box::new(Error::new(
+                ErrorKind::NotFound,
+                "No symbol found for position",
+            ))),
         }
+    }
 
-        let output = String::from_utf8(command_result.stdout)?;
+    pub async fn get_file_symbols(
+        &self,
+        file_name: &str,
+    ) -> Result<Vec<AstGrepMatch>, Box<dyn std::error::Error>> {
+        self.scan_file(&self.symbol_config_path, file_name).await
+    }
 
-        let mut symbols: Vec<AstGrepMatch> =
-            serde_json::from_str(&output).map_err(|e| format!("Failed to parse JSON: {}", e))?;
-        symbols = symbols
+    pub async fn get_file_identifiers(
+        &self,
+        file_name: &str,
+    ) -> Result<Vec<AstGrepMatch>, Box<dyn std::error::Error>> {
+        self.scan_file(&self.identifier_config_path, file_name)
+            .await
+    }
+
+    pub async fn get_references_contained_in_symbol(
+        &self,
+        file_name: &str,
+        symbol: &Symbol,
+        full_scan: bool,
+    ) -> Result<Vec<AstGrepMatch>, Box<dyn std::error::Error>> {
+        let full_path = get_mount_dir().join(&file_name);
+        let full_path_str = full_path.to_str().unwrap_or_default();
+
+        // Get all references
```

Comment:
I think references is a loaded term, I'm not sure what exactly this makes - requesting a comment/docstring